client/blocks/import/capture/capture-input.tsx

```diff
@@ -31,6 +32,20 @@ const CaptureInput: FunctionComponent< Props > = ( props ) => {
 	const [ submitted, setSubmitted ] = useState( false );
 	const exampleInputWebsite = 'www.artfulbaker.blog';
 	const showValidationMsg = hasError || ( submitted && ! isValid );
+	const { search } = useLocation();
+
+	useEffect( () => {
+		const urlValue = new URLSearchParams( search ).get( 'from' ) || '';
+		if ( urlValue ) {
+			const isValid = CAPTURE_URL_RGX.test( urlValue );
+			if ( isValid ) {
+				onInputEnter( urlValue );
+				setSubmitted( true );
+			} else {
+				setUrlValue( urlValue );
+			}
+		}
```

Comment:
I suggest moving this block of code into a separate function and naming it correctly, so we can easily understand what code do by the function name.

For example:
```useEffect( () => { checkInitSubmissionState() } )```